src/metabase/models/card.clj

```diff
@@ -274,6 +274,49 @@
   pre-update-check-sandbox-constraints
   (atom identity))
 
+(defn- update-parameter-use-card-as-sources
+  "Update the config of parameter on any Dashboard/Card use this `card` as values source .
+
+  Remove parameter.values_source_type and set parameter.values_source_type to nil ( the default type ) when:
+  - card is archived
+  - card.result_metadata changes"
+  [{id :id, :as changes}]
+  (let [parameter-cards   (db/select ParameterCard :card_id id)
+        deleted-param-ids (atom #{})]
```

Comment:
I can't see where `deleted-param-ids` is being updated.

Reply:
ah right, it was no longer needed. The ParameterCard will be automatically removed on updating Dashboard/Card parameters

Reply:
Removed in 4b130f5aeb4ec5845af5fdd5be83ebd2095ab62a